### ISSUE TITLE
fails with The exact Name is not in scope

### ISSUE THREAD

**Issue by AlistairB** [fused-effects/fused-effects-th#1]
Hi,

This is failing for me. See https://github.com/AlistairB/fused-simple-repro/pull/2 .

Using `stack build --ghc-options=-ddump-splices` on this branch gives the following error:

```
/home/blah/dev/fused-simple-repro/src/AppEventEmit.hs:22:1-36: Splicing declarations
    makeSmartConstructors ''AppEventEmit

  ======>
/home/blah/dev/fused-simple-repro/src/AppEventEmit.hs:22:1: error:
    emitAppEvent ::
    The exact Name ‘m_a192’ is not in scope
      forall (m_a194 :: Type) sig.
      Probable cause: you used a unique Template Haskell name (NameU), 
      Has (AppEventEmit) sig m_a192 => String -> m_a192 ()
      perhaps via newName, but did not bind it
    emitAppEvent a = send (EmitAppEvent a)
      If that's it, then -ddump-splices might be useful
    {-# INLINABLE emitAppEvent #-}
   |
22 | makeSmartConstructors ''AppEventEmit
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/blah/dev/fused-simple-repro/src/AppEventEmit.hs:22:1: error:
    The exact Name ‘m_a192’ is not in scope
      Probable cause: you used a unique Template Haskell name (NameU), 
      perhaps via newName, but did not bind it
      If that's it, then -ddump-splices might be useful
   |
22 | makeSmartConstructors ''AppEventEmit
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by patrickt** [comment]
Can you show a simulation of what the effect type looks like? Don’t have much to go on from just the type error.

**Comment by AlistairB** [comment]
The branch in the referenced PR should be enough to reproduce the issue. Specifically compiling https://github.com/AlistairB/fused-simple-repro/blob/4dfdde6b32638c5237de6385072ea1859bd3a21e/src/AppEventEmit.hs should do it? Not sure what you mean by simulation.

**Comment by patrickt** [comment]
Ahhh sorry I missed the example repo link! My bad my bad. I will dig into this!

**Comment by patrickt** [comment]
Hmm not sure what this is caused by. Maybe the explicit kind signature on `m`? Digging.

**Comment by patrickt** [comment]
Fixed in v0.1.0.1. Thank you so much for the report, @AlistairB!
